src/services/completions.ts

```diff
@@ -5516,14 +5516,18 @@ function isModuleSpecifierMissingOrEmpty(specifier: ModuleReference | Expression
     return !tryCast(isExternalModuleReference(specifier) ? specifier.expression : specifier, isStringLiteralLike)?.text;
 }
 
-function getVariableOrParameterDeclaration(contextToken: Node | undefined) {
+function getVariableOrParameterDeclaration(contextToken: Node | undefined, location: Node) {
     if (!contextToken) return;
 
-    const declaration = findAncestor(contextToken, node =>
+    const possiblyParameterDeclaration = findAncestor(contextToken, node =>

```

Comment:
I know that the existing code already uses `findAncestor`, but is this really needed @DanielRosenwasser? Can't we restrict the scope of this search?

Reply:
Do you mean do we really need two walks? I'm not sure; we could probably rewrite this to only do the second walk if the first fails though.

Reply:
Another way to do this is to check whether the `contextToken` ever becomes the `location`, and only then to start checking for variable declarations.

Reply:
> Do you mean do we really need two walks? I'm not sure; we could probably rewrite this to only do the second walk if the first fails though.

I'm not sure I'm following initial request here. Why we can't have two walks here? (since it uses existing ast and we also do it only once per completionInfo request)